parquet-hadoop/src/test/java/org/apache/parquet/hadoop/codec/TestLz4RawCodec.java

```diff
@@ -0,0 +1,123 @@
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Random;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+public class TestLz4RawCodec {
+  @Test
+  public void testBlock() throws IOException {
+    // Reuse the lz4 objects between test cases
+    Lz4RawCompressor compressor = new Lz4RawCompressor();
+    Lz4RawDecompressor decompressor = new Lz4RawDecompressor();
+
+    testBlockCompression(compressor, decompressor, "");
+    testBlockCompression(compressor, decompressor, "FooBar");
+    testBlockCompression(compressor, decompressor, "FooBar1FooBar2");
+    testBlockCompression(compressor, decompressor, "FooBar");
+    testBlockCompression(compressor, decompressor, "ablahblahblahabcdef");
+    testBlockCompression(compressor, decompressor, "");
+    testBlockCompression(compressor, decompressor, "FooBar");
+  }
+
+  // Test lz4 raw compression in the block fashion
+  private void testBlockCompression(Lz4RawCompressor compressor, Lz4RawDecompressor decompressor,
+                                    String data) throws IOException {
+    compressor.reset();
+    decompressor.reset();
+
+    int uncompressedSize = data.length();
+    byte[] uncompressedData = data.getBytes();
+
+    assert (compressor.needsInput());
+    compressor.setInput(uncompressedData, 0, uncompressedSize);
+    assert (compressor.needsInput());
+    compressor.finish();
+    assert (!compressor.needsInput());
+    assert (!compressor.finished() || uncompressedSize == 0);
+    byte[] compressedData = new byte[1000];
+
+    int compressedSize = compressor.compress(compressedData, 0, 1000);
+    assert (compressor.finished());
+
+    assert (!decompressor.finished());
+    assert (decompressor.needsInput());
+    decompressor.setInput(compressedData, 0, compressedSize);
+    assert (!decompressor.finished());
+    byte[] decompressedData = new byte[uncompressedSize];
+    int decompressedSize = decompressor.decompress(decompressedData, 0, uncompressedSize);
+    assert (decompressor.finished());
+
+    assertEquals(uncompressedSize, decompressedSize);
+    assertArrayEquals(uncompressedData, decompressedData);
+  }
+
+  // Test lz4 raw compression in the streaming fashion
+  @Test
+  public void testCodec() throws IOException {
+    Lz4RawCodec codec = new Lz4RawCodec();
+    Configuration conf = new Configuration();
+    int[] bufferSizes = {128, 1024, 4 * 1024, 16 * 1024, 128 * 1024, 1024 * 1024};
+    int[] dataSizes = {0, 1, 10, 1024, 1024 * 1024};
+
+    for (int i = 0; i < bufferSizes.length; i++) {
+      conf.setInt(Lz4RawCodec.BUFFER_SIZE_CONFIG, bufferSizes[i]);
+      codec.setConf(conf);
+      for (int j = 0; j < dataSizes.length; j++) {
+        testLz4RawCodec(codec, dataSizes[j]);
+      }
+    }
+  }
+
+  private void testLz4RawCodec(Lz4RawCodec codec, int dataSize) throws IOException {
+    byte[] data = new byte[dataSize];
+    (new Random()).nextBytes(data);
```

Comment:
This produces basically uncompressible data? That seems a bit insufficient.
To make the data more compressible, you can for example repeat the same chunk N times.

Reply:
Added cases to repeat chunks.